File: FitProj/app/src/main/java/com/teuskim/fitproj/BaseFragment.java
package com.teuskim.fitproj;

import android.support.v4.app.Fragment;
import android.widget.Toast;

/**
 * 모든 프레그먼트의 기반이 되는 프레그먼트
 */
public class BaseFragment extends Fragment {

    @Override
    public void onStart() {
        super.onStart();
        getView().setClickable(true);
    }

    protected void openFragment(BaseFragment fr) {
        try {
            ((MainActivity)getActivity()).openFragment(fr);
        } catch (Exception e) {}
    }

    protected void finish() {
        try {
            ((MainActivity)getActivity()).closeFragment(this);
        } catch (Exception e) {}
    }

    protected void showToast(int resId) {
        Toast.makeText(getActivity(), resId, Toast.LENGTH_SHORT).show();
    }

    protected int getStatusBarColor() {
        return getResources().getColor(R.color.bg_notibar);
    }

}

File: FitProj_kotlin/app/src/main/java/com/teuskim/fitproj/BaseFragment.kt
package com.teuskim.fitproj

import android.support.v4.app.Fragment
import android.widget.Toast

/**
 * 모든 프레그먼트의 기반이 되는 프레그먼트
 */
open class BaseFragment : Fragment() {

    open var statusBarColor: Int = 0
        get() = resources.getColor(R.color.bg_notibar)

    override fun onStart() {
        super.onStart()
        view!!.isClickable = true
    }

    protected fun openFragment(fr: BaseFragment) {
        try {
            (activity as MainActivity).openFragment(fr)
        } catch (e: Exception) {
        }

    }

    open fun finish() {
        try {
            (activity as MainActivity).closeFragment(this)
        } catch (e: Exception) {
        }

    }

    protected fun showToast(resId: Int) {
        Toast.makeText(activity, resId, Toast.LENGTH_SHORT).show()
    }

}
